core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -228,6 +240,11 @@ public static RecordState forId(byte id) {
      */
     private int stateEpoch;
 
+    /**
+     * The leader epoch is used to track the partition epoch.
+     */
+    private int leaderEpoch;
```

Comment:
Should this be volatile since it's written and read by different threads?

Reply:
It's now read only in share partition.